### ISSUE TITLE
Proposal: Migrate docs from Jekyll to MkDocs + Material for MkDocs

### ISSUE THREAD

**Issue by nscuro** [DependencyTrack/dependency-track#1842]
### Current Behavior:

We currently use [Jekyll](https://jekyllrb.com/) to build our documentation.

Jekyll is a generic static site generator and was never really intended for technical documentation, but for websites and blogs.

Although the template being used provides some basic features (syntax highlighting, search), I feel like it looks a bit dated and is missing some features that'd make both writing and reading our docs more appealing.

Jekyll is also painful to setup locally (requires Ruby, Ruby Gems, Bundler, building of native extensions).

### Proposed Behavior

What I'm proposing is the following:

* Migrate documentation from Jekyll to [MkDocs](https://www.mkdocs.org/)
* Use [Material for MkDocs](https://squidfunk.github.io/mkdocs-material/) to make the docs more visually appealing

MkDocs is written in Python and can be installed with `pip`, doesn't need native extensions and is made for writing technical documentation. Like Jekyll, it lets us write in Markdown (which can be [extended](https://squidfunk.github.io/mkdocs-material/setup/extensions/python-markdown-extensions/) in multiple ways) and can be deployed to GitHub pages and supports custom domains.

[Linking to internal pages](https://www.mkdocs.org/user-guide/writing-your-docs/#internal-links) is less painful than with Jekyll, and it has a great built-in search UX. It auto-generates table-of-contents for each page, which makes discovery and navigation a lot easier. Creating links to specific sections of a page is also very easy, which makes referring to our docs in Slack or GitHub issues a bit more practical.

Material for MkDocs provides a more modern look, ships with loads of advanced features and is highly customizable.

Because so many things are built-in, we won't have to deal with many scripts or stylesheets anymore.

For reference, here are some examples I found that also use this stack:

* [Material for MkDocs](https://squidfunk.github.io/mkdocs-material/getting-started/) itself
* [SAP Mobile Services](https://help.sap.com/doc/f53c64b93e5140918d676b927a3cd65b/Cloud/en-US/docs-en/guides/index.html)
* [AquaSec Trivy](https://aquasecurity.github.io/trivy/dev/docs/)
* [Traefik](https://doc.traefik.io/traefik/) (although heavily modified)

**Comment by nscuro** [comment]
Threw a quick POC together and migrated a few pages here: https://nscuro.github.io/dependency-track/4.6.0-SNAPSHOT/

Docs are automatically deployed using GitHub Actions: https://github.com/nscuro/dependency-track/blob/mkdocs/.github/workflows/deploy-docs.yml

Using [`mike`](https://github.com/jimporter/mike), we can serve multiple versions of the documentation: https://github.com/nscuro/dependency-track/tree/gh-pages

**Comment by ab12gu** [comment]
yeah but if you wanna just render it locally and not build everything, you could just use a md editor/viewer for jekyll. 

but i do agree deployment and visualization on local is much more annoying compared to mkdocs

**Comment by ab12gu** [comment]
@sahibamittal also this issue should be closed

**Comment by nscuro** [comment]
Implemented: https://github.com/DependencyTrack/docs
